File: swift/anagram/Sources/Anagram/Anagram.swift
class Anagram {
  let word: String

  init(word: String) {
    self.word = word
  }

  func transform(_ str: String) -> String {
    String(str.lowercased().sorted())
  }

  func match(_ candidates: [String]) -> [String] {
    return candidates.filter {transform($0) == transform(self.word) && $0.lowercased() != self.word.lowercased()}
  }
}

File: kotlin/anagram/src/main/kotlin/Anagram.kt
class Anagram(val source: String) {
    fun standarize(input: String): String {
        return input
                .toLowerCase()
                .toList()
                .sorted()
                .joinToString(separator = "")
    }

    fun match(anagrams: Collection<String>): Set<String> {
        return anagrams.filter { 
            source.toLowerCase() != it.toLowerCase() && standarize(source) == standarize(it) 
        }.toSet()
    }
}
